services/repository/files/commit.go

```diff
@@ -44,6 +45,12 @@ func CreateCommitStatus(ctx context.Context, repo *repo_model.Repository, creato
 		return fmt.Errorf("NewCommitStatus[repo_id: %d, user_id: %d, sha: %s]: %v", repo.ID, creator.ID, sha, err)
 	}
 
+	if status.State.IsSuccess() {
+		if err := automerge.MergeScheduledPullRequest(ctx, sha, repo); err != nil {
```

Comment:
Do I miss something, or does this method not ensure that the SHA points to the latest branch commit?
Because if that is the case, the implementation is incorrect at the moment as it won't do anything (as it should), but it will die with an irrelevant and confusing error...

Reply:
MergeScheduledPullRequest will trigger `getPullRequestsByHeadSHA` that only returns pulls witch have sha point to the head of a pull, all other's are ignored

Reply:
Ah, to me it looked as if it would throw an error would be thrown.
Then you can resolve this comment.